### PR TITLE
CRAFTERCMS-2529

### PR DESCRIPTION
**Pull request by dejan-brkic**
switched authentication chain configuration to a single file

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/2529

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v2/service/security/AuthenticationProviderFactory.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.craftercms.studio.api.v2.service.security.AuthenticationProvider;
+
+import java.util.Map;
+
+public class AuthenticationProviderFactory {
+
+    public static AuthenticationProvider getAuthenticationProvider(Map<String, Object> providerConfig) {
+        AuthenticationProvider provider;
+        switch (providerConfig.get("provider").toString()) {
+            case "DB":
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/security/AuthenticationProviderFactory.java]: Switch this and the others to an enum or constant.

File: src/main/java/org/craftercms/studio/impl/v2/service/security/AuthenticationProviderFactory.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                break;
+            default:
+                provider = null;
+        }
+        return provider;
+    }
+
+    private static DbAuthenticationProvider createDbAuthenticationProvider(Map<String, Object> providerConfig) {
+        DbAuthenticationProvider provider = new DbAuthenticationProvider();
+        boolean enabled = Boolean.parseBoolean(providerConfig.get("enabled").toString());
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/security/AuthenticationProviderFactory.java]: Switch literal string to a constant.

File: src/main/java/org/craftercms/studio/impl/v2/service/security/AuthenticationProviderFactory.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        provider.setEnabled(enabled);
+        return provider;
+    }
+
+    private static LdapAuthenticationProvider createLdapAuthenticationProvider(Map<String, Object> providerConfig) {
+        LdapAuthenticationProvider provider = new LdapAuthenticationProvider();
+        boolean enabled = Boolean.parseBoolean(providerConfig.get("enabled").toString());
+        provider.setEnabled(enabled);
+        provider.setLdapUrl(providerConfig.get("ldapUrl").toString());
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/security/AuthenticationProviderFactory.java]: Switch all literal strings to constants.

File: src/main/resources/crafter/studio/studio-config.yaml [hunk 4]
```diff
@@ -227,9 +227,45 @@ studio.authentication.headers.logout.enabled: false
 # be called after local logout. The {baseUrl} macro is provided so that the browser is redirected back to Studio
 # after logout (https://STUDIO_SERVER:STUDIO_PORT/studio)
 studio.authentication.headers.logout.url: /mellon/logout?ReturnTo={baseUrl}
-# Studio authentication chain configuration location
-studio.authentication.chain.configLocation: crafter/studio/authentication-chain-config.yaml
-
+# Studio authentication chain configuration
+studio.authentication.chain:
+  - provider: HEADERS
```
Comment (sumerjabri) [inline src/main/resources/crafter/studio/studio-config.yaml:233]: Just a question: Should this be call-caps? Should it be case-sensitive? Let's discuss.

File: src/main/resources/crafter/studio/studio-config.yaml [hunk 5]
```diff
@@ -227,9 +227,45 @@ studio.authentication.headers.logout.enabled: false
 # be called after local logout. The {baseUrl} macro is provided so that the browser is redirected back to Studio
 # after logout (https://STUDIO_SERVER:STUDIO_PORT/studio)
 studio.authentication.headers.logout.url: /mellon/logout?ReturnTo={baseUrl}
-# Studio authentication chain configuration location
-studio.authentication.chain.configLocation: crafter/studio/authentication-chain-config.yaml
-
+# Studio authentication chain configuration
+studio.authentication.chain:
```
Comment (sumerjabri) [inline src/main/resources/crafter/studio/studio-config.yaml:231]: I want documentation for every field and every section. Let's discuss and maybe do it together.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
See inline comments/issues. Please also fix Codacy issues.

**Review by sumerjabri — APPROVED** [review]

**Review by avasquez614 — APPROVED** [review]

**Pull request closed without merging**